src/components/CheckBox.js

```diff
@@ -16,22 +15,40 @@ type Props = {
   style?: ViewStyleProp
 };
 
-const CheckBox = ({ checked, label, onChange, style }: Props) => (
-  <Touchable
-    accessibilityComponentType="none"
-    accessibilityTraits={["none"]}
-    accessibilityLabel={checkboxAccessibilityLabel(label, checked)}
-    onPress={onChange}
-    style={[styles.container, style]}
-  >
-    <Text style={styles.text}>{label}</Text>
-    <Image source={checked ? checkBoxCheckedUrl : checkboxUrl} />
-  </Touchable>
-);
-
-CheckBox.defaultProps = {
-  style: {}
-};
+class CheckBox extends React.Component<Props> {
+  static defaultProps = {
+    style: {}
+  };
+
+  shouldComponentUpdate(nextProps: Props) {
```

Comment:
We currently use CheckBox only in the filter modal and we use an inline arrow function for `onChange`. This means all check boxes in the list are always rerendered. The easiest fix is to use a custom `shouldComponentUpdate` here.